grammar.js

```diff
@@ -215,7 +215,28 @@ module.exports = grammar({
       )
     )),
 
+    _literal: $ => choice(
+      $.symbol,
+      $.integer,
+      $.float,
+      $.boolean,
+      $.nil,
+      $.string
+    ),
+
     symbol: $ => token(seq(':', choice(identifierChars(), operatorChars()))),
+    integer: $ => (/0b[01](_?[01])*|0[oO]?[0-7](_?[0-7])*|(0d)?\d(_?\d)*|0x[0-9a-fA-F](_?[0-9a-fA-F])*/),
+    float: $ => (/\d(_?\d)*\.\d(_?\d)*([eE]\d(_?\d)*)?/),
+    boolean: $ => choice('true', 'false', 'TRUE', 'FALSE'),
+    nil: $ => choice('nil', 'NIL'),
+
+    string: $ => choice(
+      $._single_quoted,
+      $._double_quoted
+    ),
+    _single_quoted: $ => (/'(\\.|[^\\'])*'/),
+    _double_quoted: $ => token(seq('"', repeat(choice(/\\./, $.interpolation, /[^\\"]/)), '"')),
+    interpolation: $ => seq('#{', $._expression, '}'),
```

Comment:
Alternatively, could we add a combinator which turns off comment/whitespace parsing within it?